Tasks/SonarQubePostTest/PRCA/ReportProcessor.ps1

```diff
@@ -2,6 +2,80 @@ $ProjectGuidAndFilePathMap = @{}
 $ComponentKeyAndPathMap = @{}
 $ComponentKeyAndRelativePathCache = @{}
 
+
+#region Public
+
+#
+# Fetches the new issues from the json report and also adds a "relativePath" for easier consumption
+#
+# Remark: the SonarQube report uses the component guid to refer to paths
+#
+function FetchAnnotatedNewIssues
+{    
+    $sonarReportFilePath = GetSonarReportFilePath
+    if (![System.IO.File]::Exists($sonarReportFilePath))
+    {
+        throw "Could not find the SonarQube issue report at $sonarReportFilePath. Unable to post issues to the PR."
+    }
+
+    Write-Verbose "Report file found at $sonarReportFilePath"    
+        
+    CreateProjectGuidAndPathMap    
+    
+    $sonarReportFilePath = GetSonarReportFilePath
+    $json = DeserializeReport $sonarReportFilePath
+    ConstructComponentKeyAndPathMap $json
+
+    # '@' makes sure the result set is returned as an array
+    $newIssues = @($json.issues | Where {$_.isNew -eq $true})
+    Write-Host "Found $($json.issues.Count) issues out of which $($newIssues.Count) are new"
+
+    if ($($newIssues.Count) -gt 0)
+    {
+        foreach ($issue in $newIssues)
+        {
+            $filePath = GetRelativeFilePath $($issue.component)
+
+            # Add a new property in the object which stores the file path so it can be consumed directly
+            Add-Member -InputObject $issue -MemberType NoteProperty -Name "relativePath" -Value $filePath
+        }
+    }
+
+    return $newIssues
```

Comment:
I am returning the new issues as an object instead of creating another json file since the Orchestrator can now simply pass this object along between the ReportProcessor and the IssuesProcessor (not in this PR)